src/lib/pci_virtio_sock.c

```diff
@@ -434,6 +441,25 @@ static size_t iovec_clip(struct iovec **iov, int *iov_len, size_t bytes)
 	return ret;
 }
 
+static size_t iovec_shift_left1(struct iovec **iov, int *iov_len)
```

Comment:
What memory is shared here and with whom?

The control byte is something in the protocol you've designed, due to the comment where I asked about len > 0 and the interaction with the control and data parts of the message? (i.e. you need to send some dummy data to get the control message through?) If so that's definitely worth spelling out somewhere.